### ISSUE TITLE
Alignment of matrix room membership with community membership

### ISSUE THREAD

**Issue by techsmyth** [cherrytwist/Server#1496]
## Description
As a user I want that my matrix room memberships are aligned with the community membership so that I receive the right notificaitons / subscriptions

## Acceptance criteria

- [ ] User membership of matrix rooms should be kept in sync with the user membership of a community
- [ ] Admin utilities / other means for detecting when things are out of sync?

## Additional Context
Description of the different ways we interact with the Matrix rooms: https://hackmd.io/lxUr1IkeSpqWgB319CA_Hg

## Background
We are currently adding users to rooms way too often / not selective enough.

There are several problems at the moment:
- If a user has read access to a community, then the act of reading the updates or a discussion (displaying in the UI) results in the user being added to the room
- If a user is removed from a community then the user is not currently removed from the matrix rooms for that community
- we are forever checking if a user has access to all needed rooms (on *all* read requests) which is a performance / latency nightmare
- when we have the matrix client operational then the user would be a part of way too many rooms
- users are not added to rooms until they try to access the room once, so they will not receive notifications on e.g. updates

In short what we have now re matrix room membership needs a serious makeover. We need to fix this.
